### DISCUSSION TITLE
Code Bubbles

### DISCUSSION THREAD

**Discussion by cypressf** [zed-industries/zed#6455] [Feature Requests]
I have always wanted an editor to support the UI concepts in the code bubbles research project https://www.youtube.com/watch?v=PsPX0nElJ0k

Zed has a really strong technical foundations, so I thought it might be a good candidate to make a performant code-bubbles editor. What does the team think about it? Have you all thought about different UI paradigms besides file-based editing.

**Comment by tml** [comment]
I am certainly not an authority here, but from the little reading I've done on the structure of the GPUI codebase, I think that this kind of approach would require a fundamentally different architecture.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
